site/en/docs/extensions/mv3/manifest/content_scripts/index.md

```diff
@@ -0,0 +1,262 @@
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.example.com/*"],
+      "exclude_globs": ["*science*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://history.example.com
+https://.example.com/music
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://science.example.com
+https://www.example.com/science
+```
+{% endCompare %}
+</div>
+
+#### Advanced customization example {: #all-custom }
+
+{% Label %}manifest.json{% endLabel %}
+
+```json/5-7
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.example.com/*"],
+      "exclude_matches": ["*://*/*business*"],
+      "include_globs": ["*example.com/???s/*"],
+      "exclude_globs": ["*science*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://www.example.com/arts/index.html
+https://.example.com/jobs/index.html
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://science.example.com
+https://www.example.com/jobs/business
+https://www.example.com/science
+```
+{% endCompare %}
+</div>
+
+{% endDetails %}
+
+## Frames {: #frames }
+
+The `"all_frames"` field specifies if the content script should be injected into all frames matching the specified URL requirements. If set to `false` it will only inject into the topmost frame. It can be used along with `"match_about_blank"` to inject into an `about:blank` frame. 
+
+To inject into other frames like `data:`, `blob:`, and `filesystem:`, set the `"match_origin_as_fallback"` to `true`. For details, see [Injecting in related frames][cs-inject-frames]
+
+`"all_frames"` Boolean
+: _Optional_. Defaults to `false`, meaning that only the top frame is matched. If set to true, it will inject into all frames, even if the frame is not the topmost frame in the tab. Each frame is checked independently for URL requirements, it won't inject into child frames if the URL requirements are not met.
+
+`"match_about_blank"`- Boolean
+: _Optional_. Defaults to `false`. Whether the script should inject into an `about:blank` frame where the parent URL matches one of the patterns declared in `"matches"`.
+
+`"match_origin_as_fallback"` - Boolean
+: _Optional_. Defaults to `false`. Whether the script should inject in frames that were created by a matching origin, but whose URL or origin may not directly match the pattern. These include frames with different schemes, such as `about:`, `data:`, `blob:`, and `filesystem:`.
+
+## Run time and execution environment {: #world-timings }
+
+By default, content scripts are injected when the document and all resources are finished loading (`"document_idle"`), and live in a private isolated execution environment that isn't accessible to the page or other extensions.
+
+[`"run_at"`][scripting-runat] - `document_start` | `document_end` | `document_idle`
```

Comment:
Shouldn't the possible values be in quotes?

Reply:
Since you define these below, I don't think you need them here. If you want them scannable, make the items below into bullets.